nemo_curator/utils/distributed_utils.py

```diff
@@ -343,13 +357,81 @@ def read_single_partition(
     else:
         df = read_f(files, **read_kwargs, **kwargs)
 
-    if filetype in ["jsonl", "json"] and columns is not None:
-        if add_filename and "filename" not in columns:
-            columns.append("filename")
-        df = df[columns]
+    print(f"Reading with {read_kwargs=}", flush=True)
+    return select_and_sort_columns(df, columns, add_filename)
 
-    df = df[sorted(df.columns)]
-    return df
+
+def read_data_cudf_blocksize(
+    input_files: List[str],
+    file_type: Literal["parquet", "jsonl"],
+    blocksize: str,
+    add_filename: bool = False,
+    input_meta: Union[str, dict] = None,
+    columns: Optional[List[str]] = None,
+    **kwargs,
+) -> dask_cudf.DataFrame:
+    import dask_cudf
+
+    read_kwargs = dict()
+    if file_type == "jsonl":
+        read_func = dask_cudf.read_json
+        read_kwargs["lines"] = True
+        if input_meta is not None:
+            read_kwargs["prune_columns"] = True
+            read_kwargs["dtype"] = (
+                ast.literal_eval(input_meta)
+                if isinstance(input_meta, str)
+                else input_meta
+            )
+        if add_filename:
+            read_kwargs["include_path_column"] = add_filename
+
+    elif file_type == "parquet":
+        if add_filename:
+            msg = "add_filename and blocksize cannot be set at the same time for parquet files"
+            raise ValueError(msg)
```

Comment:
I think it would be nice to look further into Parquet support for `include_path_column`. In the meantime, is there any way we could support `add_filename` too? It's an unfortunate trade-off here otherwise.

Reply:
Yea, I've raised it with the `dask_cudf` team, once we get consensus we can create a feature request to support  https://github.com/dask/dask/pull/9441